src/mailroom_madness.py

```diff
@@ -0,0 +1,267 @@
+# -*- coding: utf-8 -*-
+"""File contains a mailroom helper function."""
+from __future__ import division
+import sys
+
+
+# backwards compatible input
+try:
+    input = raw_input
+except NameError:
+    pass
+
+
+def prompt_donor_name(donor_data):
+    """Function gets name from stdin.
+
+    If name is list, list donor_data.
+    If name is not in donor_data, enter it with [] value.
+    Returns name.
+
+    Args:
+        donor_data (dict)
+    """
+    name = get_name(donor_data)
+    if not name:
+        return False
+    donor_data.setdefault(name, [])
+    return name
+
+
+def get_name(donor_data, out=sys.stdout):
+    """Get a name from the user.
+
+    Return the amount or false if the
+    user wishes to return to the main prompt.
+
+    Args:
+        donor_data (dict)
+        out (file, optional)
+    """
+    prompt = u'Enter a donor name, "list", or "cancel" > '
+    while True:
+        name = input(prompt)
+        if name == u'list':
+            list_donor_names(donor_data.keys(), out)
+        elif name == u'cancel':
+            return False
+        elif name:
+            return name
+        else:
+            out.write('Please give a name.')
+
+
+def list_donor_names(names, out=sys.stdout):
+    """Print a list of donor names.
+
+    Args:
+        names (list): List of donor names.
+        out (file, optional)
+    """
+    out.write(u'Donors:\n')
+    for name in sorted(names):
+        out.write(u'* {}\n'.format(name))
+
+
+def get_donation_amount(out=sys.stdout):
+    """Get an donation amount from the user.
+
+    Return the amount or
+    false if the user wishes to return to the main prompt.
```

Comment:
I like the way your `get_*` functions all share a behavior.  That's a good sign.